meilisearch-release-changelog/v1.2.0.md

```diff
@@ -0,0 +1,208 @@
+- Descending sort is now as performant as ascending sort.
+
+### Relevancy improvements
+
+- Synonyms are no longer considered as exact matches by the `exactness` ranking rule. This change is due to the goal of the `exactness` ranking rule of boosting the documents that contain the query exactly as typed by the user. Other ranking rules will still consider synonyms. Solves https://github.com/meilisearch/meilisearch/issues/3410
+- `Attributes`, `Exactness`, `typo` and `proximity` ranking rules now imply `words` if `words` was not already specified before them. This change was made because the behavior of applying these ranking rules before or without `words` was unintuitive and could hurt relevancy.
+- Split words are now looked for for digrams. This means that the query `whit ehorse` may now match documents containing `white horse`, which improves the relevancy of typo tolerance.
+- ngrams and split words are now disadvantaged in comparison to the exact word in the `typo` ranking rule.
+- The "iterative" version of ranking rules are now removed (https://github.com/meilisearch/meilisearch/issues/3356). The iterative version of algorithms would be applied depending on internal factors and would sometimes differ greatly in relevancy from the base ranking rule, leading to inconsistent relevancy.
+- Exactness will not prioritize documents containing exact words which were previously removed by the `words` ranking rule anymore. For example, when using the query `Batman: the dark knight rises` with the default term matching strategy (`last`), documents with a title starting with `Batman: the dark knight` will now be returned before documents that contain additional terms like `Batman: the dark knight crashes`, which improves relevancy. Congratulations if you read so far. Similarly, with the query: `Batman the supercalifragilisticexpialidocious knight rises`. Then the `Last` term matching strategy will reduce the query to `Batman the`. Exactness will then only prioritise documents that contain the exact words `Batman` and `the` but not `supercalifragilisticexpialidocious`, `knight`, or `rises`.
+
+### Unblocking features
+
+v1.2 brings a massive refactor of the search engine that unblocks the implementation of user-requested features in the future:
+
+- Separating the `attribute` and `position` ranking rules: https://github.com/meilisearch/meilisearch/issues/3222
+- Limiting the search to specific attributes (`words in [title, genres]`): https://github.com/meilisearch/product/discussions/595
+- More term matching strategies: https://github.com/meilisearch/meilisearch/issues/3424
+- Setting ranking rule at search time: https://github.com/meilisearch/meilisearch/issues/3425
+- Scoring search results: https://github.com/meilisearch/product/discussions/594
+
+Done by @loiclec, @dureuill, @ManyTheFish, @Kerollmops and @irevoire in #3542.
+
+## Prevent Meilisearch to stop receiving task by removing tasks automatically
+
+TBD
```

Comment:
```suggestion
Definite fix for #3622.

We've added a limit (1M) to the maximum number of tasks the task queue can hold.
Once the task queue reaches this limit (1M of tasks are in the task queue, whatever their status is), a task deletion will be scheduled and try to delete the oldest 100k tasks.
If the 100k tasks can't be deleted due to some of them being unfinished, we will delete as many tasks as possible.
If the limit is reached, you’ll still be able to register new tasks. The engine will only stop you from adding new tasks once the hard limit of 10GiB of tasks is reached (that’s between 5M and 15M of tasks depending on your workflow).

Technically, we won't enqueue a task unless we can guarantee it will be processed right away.
If the automated task deletion doesn’t delete anything, we won't enqueue any more tasks and log a warning for the user to know that the engine is not working properly.
```